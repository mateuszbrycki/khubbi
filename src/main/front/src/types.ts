import {DateTimeFormatter, LocalDateTime, ZonedDateTime, ZoneId} from "js-joda";

export class EventDate {
    value: ZonedDateTime

    public constructor(value: ZonedDateTime) {
        this.value = value
    }

    public toString(): string {
        return this.value.toString();
    }

    public format(format: string): string {
        return this.value.format(DateTimeFormatter.ofPattern(format))
    }

    public toISODateTime(): string {
        return this.value.format(DateTimeFormatter.ISO_OFFSET_DATE_TIME);
    }

    public static now(): EventDate {
        return new EventDate(ZonedDateTime.now());
    }

    public static ofDateAndTime(date: string): EventDate {
        return new EventDate(LocalDateTime.parse(date)
            .atZone(ZoneId.UTC));
    }

}

export interface Event {
    readonly note: string;
    readonly date: EventDate;
    readonly id: string;
}

export interface User {
    readonly id: string;
    readonly email: string;
    readonly createdAt: string;
}

export class EventForms {
    public static readonly NOTE = new EventForms('Note');
    public static readonly PICTURE = new EventForms('Picture');
    public static readonly EVENT = new EventForms('Event');
    public static readonly EXPENSE = new EventForms('Expense');
    public static readonly REMINDER = new EventForms('Reminder');

    private constructor(public readonly name: string) {
    }
}