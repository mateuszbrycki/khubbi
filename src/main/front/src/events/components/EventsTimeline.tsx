import {List} from "immutable";
import {Event} from "../../types";
import React from "react";
import {Row} from "react-bootstrap";
import Container from "react-bootstrap/Container";

export interface EventsTimelineProps {
    readonly events: List<Event>
}

export interface EventsTimelineActionProps {
    readonly loadEvents: () => void
}

const EventsTimeline: React.FC<EventsTimelineProps & EventsTimelineActionProps> = (props) => {
    const {events, loadEvents} = props

    React.useEffect((): void => {
        loadEvents()
    }, [loadEvents])

    return (<Container className="justify-content-left">
        <Row>
            <h1>Events</h1>
        </Row>
        <Row>
            {events ? events.map(event => <li
                key={event.id}>{event.note} ({event.date.format("dd-MM-yyyy HH:mm")})</li>) : <></>}
        </Row>
    </Container>)

}

export default EventsTimeline