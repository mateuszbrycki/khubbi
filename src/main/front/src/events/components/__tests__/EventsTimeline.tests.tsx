import ShallowRenderer from "react-test-renderer/shallow";
import EventsTimeline from "../EventsTimeline";
import {List} from "immutable";
import {EventDate} from "../../../types";


it('renders timeline with events', () => {
    const renderer = ShallowRenderer.createRenderer()
    renderer.render(<EventsTimeline events={List.of({
        note: "event-1",
        id: "event-1",
        date: EventDate.ofDateAndTime("2024-07-12T20:00").atUTC()
    }, {
        note: "event-2",
        id: "event-2",
        date: EventDate.ofDateAndTime("2024-07-12T21:00").atUTC()
    })}
                                    loadEvents={() => null}
    />)
    expect(renderer.getRenderOutput()).toMatchSnapshot()
});