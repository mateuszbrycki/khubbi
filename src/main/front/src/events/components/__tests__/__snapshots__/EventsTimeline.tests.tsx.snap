// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders timeline with events 1`] = `
<Container
  className="justify-content-left"
>
  <Row>
    <h1>
      Events
    </h1>
  </Row>
  <Row>
    Immutable.List [
      <li>
        event-1
         (
        12-07-2024 20:00
        )
      </li>,
      <li>
        event-2
         (
        12-07-2024 21:00
        )
      </li>,
    ]
  </Row>
</Container>
`;
